pkg/ingester/retention.go

```diff
@@ -0,0 +1,225 @@
+package ingester
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/services"
+	"github.com/oklog/ulid"
+
+	"github.com/grafana/phlare/pkg/phlaredb"
+	"github.com/grafana/phlare/pkg/phlaredb/block"
+	diskutil "github.com/grafana/phlare/pkg/util/disk"
+)
+
+const (
+	defaultMinFreeDisk                        = 10 * 1024 * 1024 * 1024 // 10Gi
+	defaultMinDiskAvailablePercentage         = 0.05
+	defaultRetentionPolicyEnforcementInterval = 5 * time.Minute
```

Comment:
I think it might be worthy to make the parameters configurable

Reply:
I tend to agree, it will always be a balance between flexibility and making Pyroscope complex to operate, but being able to change that or disable deletion all together is more than a nice to have. That said I wouldn't mind if you do this in here or in a follow up PR.